zerver/actions/realm_settings.py

```diff
@@ -518,6 +523,8 @@ def do_set_realm_user_default_setting(
         property=name,
         value=value,
     )
+    if isinstance(raw_value, Enum):
+        event["value"] = raw_value.name
```

Comment:
It'd be better to instead define an `event_value` field above in your existing if/else statement.